docs/blogs/2023/2023-06-07-fast-conformer.md

```diff
@@ -0,0 +1,210 @@
+# page_path: Relative path to the image from the website root (e.g. /assets/images/)
+description: Efficient training and inference on long audio with Fast Conformer architecture
+
+# DO NOT CHANGE BELOW
+template: blog.html
+---
+
+# Fast Conformer with Linearly Scalable Attention for Efficient Speech Recognition
+
+The Conformer architecture, introduced by [Gulati et al.](https://arxiv.org/abs/2005.08100) has been a standard architecture used for not only Automatic Speech Recognition, but has also been extended to other tasks such as Spoken Language Understanding, Speech Translation, and used as a backbone for Self Supervised Learning for various downstream tasks. While they are highly accurate models on each of these tasks, and can be extended for use in other tasks, they are also very computationally expensive. This is due to the quadratic complexity of the attention mechanism, which makes it difficult to train and infer on long sequences, which are used as input to these models due to the granular stride of audio pre-processors (commonly Mel Spectrograms or even raw audio signal in certain models with 10 milliseconds stride). Furthermore, the memory requirement of quadratic attention also significantly limits the audio duration during inference.
+
+In this paper, we introduce the **Fast Conformer** architecture, which applies simple changes to the architecture to significantly reduce the computational cost of training and inference, all while mantaining the strong results of the original Conformer model. We further show that by modifying (on the fly) the global attention module to a linearly scalable attention mechanism - the same model can be used to train (or finetune) and then infer on long sequences (up to 1 hour !).
+
+Please refer to the paper here: [Fast Conformer with Linearly Scalable Attention for Efficient Speech Recognition](https://arxiv.org/abs/2305.05084)
+
+<div align="center">
+<img src="https://github.com/NVIDIA/NeMo/releases/download/v1.18.0/asset-post-fast-conformer-diagram.png">
+</div>
+
+## Fast Conformer: Architecture
+
+We propose 4 changes to the original Conformer architecture to make it more efficient:
+
+1) **Downsampling module**: The original Conformer paper uses a stack of 2-D Convolutions with a large number of output filters in order to perform the downsampling in order to reduce the resolution of the incoming audio frames from 10 mz to 40 ms, which makes it more tractable for the subsequent attention layers to operate on. However, these convolutions amount to roughly 20 % of the entire time required to perform a single forward pass of of the "Large" Conformer (with 120 M parameters). Furthermore, due to the quadratic attention cost, we can option a 4x reduction in all subsequent attention layers by downsampling the audio to 80 ms frames. So as the first change, we perform 8x downsampling before any of the Conformer layers are applied. This reduces GMACs to roughly 65% of the baseline Conformer.
```

Comment:
10mz -> 10ms

Reply:
`of of` to `of`

Reply:
we can option a 4x reduction? what does option mean here?

Reply:
Good catches !